js/src/logger.ts

```diff
@@ -1033,12 +1125,17 @@ export class ReadonlyAttachment {
     const state = this.state ?? _globalState;
     await state.login({});
 
-    const resp = await state.apiConn().get("/attachment", {
-      key: this.reference.key,
+    let params: Record<string, string> = {
       filename: this.reference.filename,
       content_type: this.reference.content_type,
       org_id: state.orgId || "",
-    });
+    };
+    if (this.reference.type === "braintrust_attachment") {
+      params.key = this.reference.key;
+    } else if (this.reference.type === "external_attachment") {
+      params.url = this.reference.url;
+    }
+    const resp = await state.apiConn().get("/attachment", params);
```

Comment:
maybe we should move this to the classes' `data()` or maybe a new `get()` (or a better name)? 

This way we avoid having to make the params set, and we avoid the check

Reply:
ill probably leave as is since it doesn't add much complexity here